genie_registry/maf.py

```diff
@@ -277,6 +277,58 @@ def _validate(self, mutationDF):
 
         return total_error.getvalue(), warning.getvalue()
 
+    def _cross_validate(self, mutationDF: pd.DataFrame) -> tuple:
+        """This function cross-validates the mutation file to make sure it
+        adheres to the mutation SOP.
+
+        Args:
+            mutationDF (pd.DataFrame): mutation dataframe
+
+        Returns:
+            Text with all the errors in the mutation file
+        """
+        errors = ""
+        warnings = ""
+        maf_filename = f"data_mutations_extended_{self.center}.txt"
+        id_to_check = "SAMPLE_ID"
+
+        clinical_files = {
+            file.name: file.path
+            for files in self.ancillary_files
+            for file in files
+            if file.name.startswith("data_clinical_supp")
+        }
+        clinical_file_names = ''.join(clinical_files.keys())
+        if clinical_files:
+            try:
+                clinical_sample_df = process_functions.get_clinical_dataframe(
+                    filePathList=list(clinical_files.values())
+                )
+            except Exception as e:
+                errors = (
+                    f"The clinical file(s) cannot be read. No cross-validation will be done. Original error: {str(e)}"
+                )
+                warnings = ""
+
+            if not errors:
+                if process_functions.checkColExist(
+                    mutationDF, id_to_check
+                ) and process_functions.checkColExist(clinical_sample_df, id_to_check):
+                    errors, warnings = self.cross_validate_ids_between_two_files(
+                        df1=mutationDF,
+                        df1_filename=maf_filename,
```

Comment:
The maf file has the sample column at `Tumor_Sample_Barcode` which is different from `SAMPLE_ID`